rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -173,10 +173,12 @@ NodeGraph::get_node_names_and_namespaces() const
     throw std::runtime_error(error_msg);
   }
 
-  std::vector<std::pair<std::string, std::string>> node_names(node_names_c.size);
+  // TODO(jhdcs): If ranges are ever added to ROS2, replace following with range-based filter-transform
```

Comment:
That would be cool, but this same desire applies to other places in the code, so maybe we just drop the TODO for now. If you really want to track it we can make an issue, but realistically we'd audit the code when we're able and replace all instances where it would be better.

Reply:
I'm just really fond of trying out functional design in areas that don't really have them before. I'll remove this TODO though.

There is a library that adds ranges on GitHub (it's actually the basis for a proposal to add ranges to the standard library), but I don't want to start importing random codebases willy-nilly without consent. _Especially_ since this isn't my project.

Reply:
Yeah, unfortunately I think pragmatism wins here (I've been known import future features as third-party packages too, e.g. `std::variant` and `std::filesystem`). Thanks for removing it.